block/internal/syncing/p2p_handler.go

```diff
@@ -16,16 +15,24 @@ import (
 	"github.com/evstack/ev-node/types"
 )
 
-// P2PHandler handles all P2P operations for the syncer
+// P2PHandler coordinates block retrieval from P2P stores for the syncer.
+// It waits for both header and data to be available at a given height,
+// validates their consistency, and emits events to the syncer for processing.
+//
+// The handler maintains a processedHeight to track the highest block that has been
+// successfully validated and sent to the syncer, preventing duplicate processing.
 type P2PHandler struct {
 	headerStore goheader.Store[*types.SignedHeader]
 	dataStore   goheader.Store[*types.Data]
 	cache       cache.Manager
 	genesis     genesis.Genesis
 	logger      zerolog.Logger
+
+	mu              sync.Mutex
```

Comment:
nit, we've removed mutexes from the whole codebase, let's use `*atomic.Uint64` instead